### PR TITLE
feat: add notification card to design system

### LINKED ISSUE #1106: Feature: Add reactions as notifications
### Type of feature

🍕 Feature

### Current behavior

This will need some API updates to identify when things need to be read. @brandonroberts can you confirm this. 

### Suggested solution

Screen Shot 2023-04-18 at 4 02 12 PM

When a user reacts to your highlight you should get a notification in the bell dropdown. For now, we can make the notification unread when a new reaction comes in and read it when a user clicks the bell. We can make it more complicated in the future.
 

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [ ] I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by OgDev-01**
## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Description
This PR adds the notification card to the design system
This is prep work to implement #1106
A new color was also included in the tailwind config file `hsl(24, 100%, 95%)`

## Mobile & Desktop Screenshots/Recordings
image

image

image

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [x] 📕 storybook
- [ ] 🙅 no documentation needed

### CODE REVIEW

File: components/atoms/NotificationsCard/notification-card.tsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import React from "react";
+import { FaRegSmile, FaUserCircle } from "react-icons/fa";
+
+interface NotificationCard {
+  type: "reaction" | "follow" | "collaboration";
+  message: string;
+}
+const NotificationCard = ({ type, message }: NotificationCard) => {
```
Comment (0-vortex) [inline components/atoms/NotificationsCard/notification-card.tsx:8]: Suggest this if the purpose is just rotating icons and not any custom design:

```
const NotificationCard = ({ type, message }: NotificationCard) => {
  const Icons = {
    reaction: FaRegSmile,
    follow: FaUserCircle,
    collaboration: FaUserCircle
  };
  
  const Icon = Icons[type];

  return (
    <div className="flex items-center gap-3 p-2 transition cursor-pointer rounded-xl group hover:text-sauced-orange hover:bg-sauced-light">
      <Icon className="text-3xl shrink-0"/>
      <p className="text-sm">{message}</p>
    </div>
  );
}
```

File: tailwind.config.js [hunk 2]
```diff
@@ -23,7 +23,8 @@ module.exports = {
       },
       colors: {
         sauced: {
-          orange: "hsla(19, 100%, 50%, 1)"
+          orange: "hsla(19, 100%, 50%, 1)",
+          light: "#FFEFE4"
```
Comment (0-vortex) [inline tailwind.config.js:27]: Suggest using tools like https://find-nearest-tailwind-colour.netlify.app to identify similar tailwind colors - for the above, the linked tool will fail linking `rose-100` as the color, however, looking at https://tailwindcss.com/docs/customizing-colors a similar enough color exists as `orange-100`

If the suggestion is not appropriate, it should be raised in the PR that new colors are being added to the design system, as the insights configuration is sometimes linked or copy pasted into other repos! 🍕

**Review by Deadreyo — APPROVED** [review]
LGTM! 🚀

File: components/atoms/NotificationsCard/notification-card.tsx (hunk 1 above)
Comment (OgDev-01) [reply components/atoms/NotificationsCard/notification-card.tsx:8]: Thanks!

File: tailwind.config.js (hunk 2 above)
Comment (OgDev-01) [reply tailwind.config.js:27]: It's a great suggestion, but several new color values are being introduced into the design system as we expand, and just to stay in uniform with what we already have in place, I can convert the color value into hsl using tools like https://htmlcolors.com/hex-to-hsl.

File: components/atoms/NotificationsCard/notification-card.tsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
```
Comment (brandonroberts) [inline components/atoms/NotificationsCard/notification-card.tsx:8]: nit: For consistency with other props interfaces

```suggestion
const NotificationCard = ({ type, message }: NotificationCardProps) => {
```

**Review by brandonroberts — CHANGES_REQUESTED** [review]

**Review by brandonroberts — APPROVED** [review]

**Pull request merged**